blurring_as_a_service/inference_pipeline/components/detect_and_blur_sensitive_data.py

```diff
@@ -44,23 +47,37 @@ def detect_and_blur_sensitive_data(
         Pre-trained model to be used to blur.
     results_path:
         Where to store the results.
+    yolo_yaml_path:
+        Where to store the yaml file which is used during validation
+
     """
-    files_to_blur_full_path = "outputs/files_to_blur_full_path.txt"
+    files_to_blur_full_path = "outputs/files_to_blur_full_path.txt"  # use outputs folder as Azure expects outputs there
     with open(relative_paths_files_to_blur, "r") as src:
         with open(files_to_blur_full_path, "w") as dest:
             for line in src:
                 dest.write(f"{mounted_root_folder}/{line}")
+                print(f"{mounted_root_folder}/{line}")
+
+    data = dict(
+        train=f"../{files_to_blur_full_path}",
+        val=f"../{files_to_blur_full_path}",
+        test=f"../{files_to_blur_full_path}",
+        nc=2,
+        names=["person", "license_plate"],
+    )
 
+    with open(f"{yolo_yaml_path}/pano.yaml", "w") as outfile:
+        yaml.dump(data, outfile, default_flow_style=False)
+    cuda_device = torch.cuda.current_device()
     model_parameters = settings["inference_pipeline"]["model_parameters"]
-    detect.run(
+    val.run(
         weights=f"{model}/best.pt",
-        source=files_to_blur_full_path,
+        data=f"{yolo_yaml_path}/pano.yaml",
         project=results_path,
-        save_txt=model_parameters["save_txt"],
-        exist_ok=model_parameters["exist_ok"],
-        name="detection_result",
+        batch_size=4,
```

Comment:
Do we want to add this in settings["inference_pipeline"]["model_parameters"]

model_parameters["batch_size"]

Reply:
yes, let's keep it there so it's easily adjustable

Reply:
Added this